nuxeo-compound-documents-web/test/nuxeo-compound-document-tree.test.js

```diff
@@ -0,0 +1,826 @@
+    levelOneDocument = levelOneDocuments[2];
+
+    // document definition for: root -> documents -> documents
+    levelTwoDocuments = [
+      generateDocument({ type: 'File', parentRef: '4', parentPath: '/Folder4', isFolderish: false }),
+      generateDocument({ type: 'Folder', parentRef: '4', parentPath: '/Folder4', isFolderish: true }),
+    ];
+    // eslint-disable-next-line prefer-destructuring
+    levelTwoDocument = levelTwoDocuments[1];
+
+    // document definition for: root -> documents -> documents -> documents
+    levelThreeDocuments = [
+      generateDocument({ type: 'File', parentRef: '6', parentPath: '/Folder6/Folder7', isFolderish: false }),
+      generateDocument({ type: 'File', parentRef: '6', parentPath: '/Folder6/Folder8', isFolderish: false }),
+    ];
+
+    // set the breadcrumb for some documents
+    levelTwoDocument.contextParameters.breadcrumb = {
+      'entity-type': 'documents',
+      entries: [levelOneDocument],
+    };
+
+    levelThreeDocuments[0].contextParameters.breadcrumb = {
+      'entity-type': 'documents',
+      entries: [levelOneDocument, levelTwoDocument],
+    };
+  }
+
+  function setupServerResponses() {
+    server.respondWith('GET', '/api/v1/path/', [200, jsonHeader, JSON.stringify(rootDocument)]);
+    server.respondWith('GET', '/api/v1/search/pp/tree_children/execute?currentPageIndex=0&pageSize=-1&queryParams=1', [
+      200,
+      jsonHeader,
+      JSON.stringify(generatePageProviderResponse(levelOneDocuments)),
+    ]);
+    server.respondWith(
+      'GET',
+      '/api/v1/search/pp/tree_children/execute?currentPageIndex=0&offset=0&pageSize=40&queryParams=1',
+      [200, jsonHeader, JSON.stringify(generatePageProviderResponse(levelOneDocuments))],
+    );
+    server.respondWith(
+      'GET',
+      '/api/v1/search/pp/tree_children/execute?currentPageIndex=0&offset=0&pageSize=40&queryParams=4',
+      [200, jsonHeader, JSON.stringify(generatePageProviderResponse(levelTwoDocuments))],
+    );
+    server.respondWith(
+      'GET',
+      '/api/v1/search/pp/tree_children/execute?currentPageIndex=0&offset=0&pageSize=40&queryParams=7',
+      [200, jsonHeader, JSON.stringify(generatePageProviderResponse())],
+    );
+    server.respondWith(
+      'GET',
+      '/api/v1/search/pp/tree_children/execute?currentPageIndex=0&offset=0&pageSize=40&queryParams=6',
+      [200, jsonHeader, JSON.stringify(generatePageProviderResponse(levelThreeDocuments))],
+    );
+    server.respondWith('GET', '/api/v1/path/Folder4', [200, jsonHeader, JSON.stringify(levelOneDocument)]);
+    server.respondWith('GET', '/api/v1/path/Folder4/Folder6', [200, jsonHeader, JSON.stringify(levelTwoDocument)]);
+  }
+
+  async function setupFixture() {
+    // create the document tree
+    documentTree = await fixture(
+      html` <nuxeo-compound-document-tree .router=${router} visible></nuxeo-compound-document-tree> `,
+      true,
+    );
+    documentTree.document = rootDocument;
```

Comment:
No, not really, I think I kept it this way because I incrementally designed the test and never moved the document to the fixture setup. I'll change it. Thanks 🙂